src/Model/Table/SentencesTable.php

```diff
@@ -352,11 +352,13 @@ public function findFilteredTranslations($query, $options) {
                     if ($translation->indirect_translations) {
                         foreach ($translation->indirect_translations as $indirectTranslation) {
                             if (!in_array($indirectTranslation->id, $indirectIds)) {
+                                $indirectTranslation = $this->newEntity($indirectTranslation->toArray());
```

Comment:
It looks like this line doesn’t do anything but duplicating the object. Is that really needed?

Reply:
Ah yes, sorry, I forgot to answer this.

It is needed because of the virtual fields (`lang_name`,  `dir`, `is_favorite`, `is_owned_by_current_user`). These virtual fields won't be set in the `contain`'ed entities... So recreating the entity was a way to have them set. If you have another solution I'm all ears.

Reply:
Okay I found the problem. The entity is not a sentence, it’s a translation, so the virtual fields have to be defined in src/Model/Entity/Translation.php.

Reply:
Thanks! I created the Translation entity and made it extend Sentence. It works fine from there.